### ISSUE TITLE
Win32 compilation issue

### ISSUE THREAD

**Issue by SpiceyWolf** [godot-dlang/godot-dlang#105]
Trying to build against 32bit using dub with the `-a x86` flag returns errors about implicit type conversions:

```
C:\users\spiceywolf\AppData\Local\dub\packages\godot-dlang-0.2.0\godot-dlang\src\godot\array.d(319,27): Error: cannot implicitly convert expression `this._array._
bind.count(((const const(Variant) __copytmp59097 = (__copytmp59097 = v).__postblit();) , __copytmp59097))` of type `long` to `uint`
C:\users\spiceywolf\AppData\Local\dub\packages\godot-dlang-0.2.0\godot-dlang\src\godot\array.d(420,26): Error: cannot implicitly convert expression `this._array._
bind.size()` of type `long` to `uint`
C:\users\spiceywolf\AppData\Local\dub\packages\godot-dlang-0.2.0\godot-dlang\src\godot\poolarrays.d(170,26): Error: cannot implicitly convert expression `this._pa
cked_array._bind.size()` of type `long` to `uint`
C:\users\spiceywolf\AppData\Local\dub\packages\godot-dlang-0.2.0\godot-dlang\src\godot\poolarrays.d(72,27): Error: template instance `godot.poolarrays.PackedArray
!string` error instantiating
```

Its a decent sized list of complaints, seemingly all from this one file, but possibly others. I don't know how this would fair against a raw 32bit machine trying to build cause I havent tested that but I assume the result would be the same.

**Comment by al1-ce** [comment]
It seems that it's tied to D alias `size_t` and godot bindings sizes

**Comment by SpiceyWolf** [comment]
Would that require them to fix in the Godot bindings then?

**Comment by al1-ce** [comment]
Ye, it's gonna require to either use godot size i guess

**Comment by Superbelko** [comment]
I know this is a bit old, but how crucial it is to have plain x86 builds?

**Comment by SpiceyWolf** [comment]
Considering Godot is a lower req engine than Unity and Unreal engine I would say as a general yes because more games made in it (or general purpose apps since godot is great for that) will end up being light enough to run there. For a personal requirement of mine in a few projects, definitely yes because a few projects I need dynamic lib loading and a few other things that gdscript isnt great at (or capable of in general) would get decently used by people buying low budget laptops and somewhat older desktops that run 32bit windows.

**Comment by Superbelko** [comment]
Fixed with #154 with support for both float/double godot builds, use version `GODOT_REAL_T_DOUBLE` with godot builds with `double` precision enabled.
